GoodFirstIssue-ToDoApp/Screens/ItemEdit/ItemEditViewController.swift

```diff
@@ -105,6 +96,29 @@ private extension ItemEditViewController {
     }
     
     @objc func saveButtonTapped(_ sender: UIBarButtonItem) {
+        do {
+            let realm = try Realm()
+            switch mode {
+            case .create:
+                let item: Item = Item()
+                var maxId: Int { return realm.objects(Item.self).sorted(byKeyPath: "id").last?.id ?? 0 }
+                item.id = maxId + 1
+                item.name = nameTextField.text!
+                try! realm.write {
+                    realm.add(item)
+                }
+            case .edit(item: currentItem):
+                currentItem!.name = nameTextField.text!
+                try! realm.write {
+                    realm.add(currentItem!, update: Realm.UpdatePolicy.all)
+                }
```

Comment:
```swift
enum Mode {
    case create
    case edit(id: Int)
}
```

としたうえで、

```suggestion
            case .edit(id: Int):
                guard let item = realm.object(ofType: Item.self, forPrimaryKey: id) else {
                    assertFailure("id is invalid")
                }
                
                try! realm.write {
                    item.name = nameTextField.text!
                }
```

とかするとシンプルにできると思います。
currentItem や itemId も要らなくなりそう。
createモード時にはidにアクセスできなくなるので安全になります。

Reply:
参考
Realm Structure Reference
https://realm.io/docs/swift/latest/api/Structs/Realm.html#/s:10RealmSwift0A0V6object6ofType13forPrimaryKeyxSgxm_q_tSo0aB6ObjectCRbzr0_lF

Reply:
```
enum Mode {
    case create
    case edit(id: Int)
}
```
に直して、currentItemは削除してみました。itemIdは削除できなかったというか、
```
case .edit(id: Int):
```
という書き方がコンパイルエラーになってしまい、
```
case .edit(id: itemId):
```
という書き方にしました。ここは `case .edit:` という書き方でもいけそうでしたが、上記の書き方を採用しました。

Reply:
あっ、間違えました。 `case let .edit(id):` ですね。

Reply:
ちなみに `case let .edit(id: itemID):` と `case .edit(id: itemID):` の違いは分かりますか？

Reply:
@akio0911 
case let .edit(id: itmeID): と
case .edit(id: itemID): の違いはよく理解できていいません。

Reply:
1. case let .edit(id: itemID): は case let .edit(id: 10): と同じで、
「.editで、付随するInt値がitemIDと同じなら以下の処理を行う」ですね。
2. case .edit(id: itemID): は
「.editなら、付随するInt値をitemIDという定数に代入して以下の処理を行う」

なので1は条件網羅できずdefaultが必要となり、2は条件網羅できているのでdefaultが不要になります。